### ISSUE TITLE
Pylance incorrectly flags “actual” function definition in group of overloaded functions as missing types

### ISSUE THREAD

**Issue by rben01** [microsoft/pyright#8548]
## Environment data

-   Pylance version: v2024.7.1
-   OS and version: macOS Sonoma 14.5
-   Python version (& distribution if applicable, e.g. Anaconda): 
    ```text
	$ poetry env info
	Virtualenv
	Python:         3.11.9
	Implementation: CPython
	Path:           /Users/rben01/Library/Caches/pypoetry/virtualenvs/base-wTklQcO3-py3.11
	Executable:     /Users/rben01/Library/Caches/pypoetry/virtualenvs/base-wTklQcO3-py3.11/bin/python
	Valid:          True
    ```

## Code Snippet

This is copied directly from the `typing` module docs, https://docs.python.org/3/library/typing.html#typing.overload

```python
@overload
def process(response: None) -> None: ...

@overload
def process(response: int) -> tuple[int, str]: ...

@overload
def process(response: bytes) -> str: ...

def process(response): ...  # actual implementation goes here
```

## Expected behavior

This code should pass the type checker; the “actual” function definition shouldn't need type annotations because that's what the `@overload`s are for.

## Actual behavior

The actual function definition is flagged as incompatible with the overloads. In the above example, the VSCode error popup has the following:

```
Overloaded implementation is not consistent with signature of overload 2
  Function return type "tuple[int, str]" is incompatible with type "None"
    "tuple[int, str]" is incompatible with "None" Pylance reportInconsistentOverload
file.py(239, 5): Overload signature is defined here
```
```
Overloaded implementation is not consistent with signature of overload 3
  Function return type "str" is incompatible with type "None"
    "str" is incompatible with "None" Pylance reportInconsistentOverload
file.py(243, 5): Overload signature is defined here
```
```
(function) def process(response: Unknown) -> None
```

## Logs

I don't see anything here but “discover tests for workspace ...”

**Comment by erictraut** [comment]
Pyright is working as designed here, so I don't consider this a bug.

If you don't want pyright to perform consistency checks between overloads and the overload implementation, you can disable the `reportInconsistentOverload` check. This check is designed for people who want to guarantee that the overload implementation signature is consistent with the overloaded signatures.
